Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -0,0 +1,247 @@
+	}
+
+	ClientRTTSeconds /= static_cast<float>(UXComponentCount) + 0.00001f; // Avoid div 0
+	ClientViewLatenessSeconds /= static_cast<float>(UXComponentCount) + 0.00001f; // Avoid div 0
+
+	AveragedClientRTTSeconds = ClientRTTSeconds;
+	AveragedClientViewLatenessSeconds = ClientViewLatenessSeconds;
+
+	if (!bHasUxFailed && AveragedClientRTTSeconds > MaxClientRoundTripSeconds && AveragedClientViewLatenessSeconds > MaxClientViewLatenessSeconds)
+	{
+		bHasUxFailed = true;
+#if !WITH_EDITOR
+		UE_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("UX metric has failed. RTT: %.8f, ViewLateness: %.8f, ActivePlayers: %d"), AveragedClientRTTSeconds, AveragedClientViewLatenessSeconds, ActivePlayers);
+#endif
+	}
+
+	PrintUXMetric -= DeltaSeconds;
+	if (PrintUXMetric < 0.0f)
+	{
+		PrintUXMetric = 10.0f;
+#if !WITH_EDITOR
+		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("UX metric values. RTT: %.8f, ViewLateness: %.8f, ActivePlayers: %d"), AveragedClientRTTSeconds, AveragedClientViewLatenessSeconds, ActivePlayers);
+#endif
+	}
+}
+
+void ABenchmarkGymGameModeBase::ParsePassedValues()
+{
+	if (FParse::Param(FCommandLine::Get(), TEXT("TotalPlayers")))
+	{
+		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Found OverrideSpawning in command line Keys, worker flags for custom spawning will be ignored."));
+		FParse::Value(FCommandLine::Get(), *TotalPlayerCommandLineKey, ExpectedPlayers);
+		FParse::Value(FCommandLine::Get(), *TotalNPCsCommandLineKey, TotalNPCs);
+		FParse::Value(FCommandLine::Get(), *MaxRoundTripCommandLineKey, MaxClientRoundTripSeconds);
+		FParse::Value(FCommandLine::Get(), *MaxLatenessCommandLineKey, MaxClientViewLatenessSeconds);
+	}
+	else
+	{
+		USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+		check(NetDriver);
+
+		UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Using worker flags to load custom spawning parameters."));
+		FString TotalPlayersString, TotalNPCsString, MaxRoundTrip, MaxViewLateness;
+
+		const USpatialWorkerFlags* SpatialWorkerFlags = NetDriver != nullptr ? NetDriver->SpatialWorkerFlags : nullptr;
+		check(SpatialWorkerFlags != nullptr);
+
+		if (SpatialWorkerFlags != nullptr)
+		{
+			if (SpatialWorkerFlags->GetWorkerFlag(TotalPlayerWorkerFlag, TotalPlayersString))
+			{
+				ExpectedPlayers = FCString::Atoi(*TotalPlayersString);
+			}
+
+			if (NetDriver->SpatialWorkerFlags->GetWorkerFlag(TotalNPCsWorkerFlag, TotalNPCsString))
+			{
+				TotalNPCs = FCString::Atoi(*TotalNPCsString);
+			}
+
+			if (NetDriver->SpatialWorkerFlags->GetWorkerFlag(MaxRoundTripWorkerFlag, MaxRoundTrip))
+			{
+				MaxClientRoundTripSeconds = FCString::Atoi(*MaxRoundTrip);
+			}
+
+			if (NetDriver->SpatialWorkerFlags->GetWorkerFlag(MaxLatenessWorkerFlag, MaxViewLateness))
+			{
+				MaxClientViewLatenessSeconds = FCString::Atoi(*MaxViewLateness);
+			}
+		}
+	}
+
+	UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Players %d, NPCs %d"), ExpectedPlayers, TotalNPCs);
```

Comment:
Realise you copied these - but can you output the value of everything possibly set in this file at the bottom here. Useful for debugging.